drivers/spi/spi_mcux_lpspi.c

```diff
@@ -276,7 +283,52 @@ static void spi_mcux_dma_callback(const struct device *dev, void *arg,
 			data->status_flags |= SPI_MCUX_LPSPI_DMA_ERROR_FLAG;
 		}
 	}
+#if CONFIG_SPI_ASYNC
+	if (data->ctx.asynchronous) {
+		const struct spi_mcux_config *config = spi_dev->config;
+
+		if ((data->status_flags & SPI_MCUX_LPSPI_DMA_DONE_FLAG)  ==
+			SPI_MCUX_LPSPI_DMA_DONE_FLAG) {
+			while ((LPSPI_GetStatusFlags(config->base) & kLPSPI_ModuleBusyFlag)) {
+			/* wait until module is idle */
+			}
+
+			/* Disable DMA */
+			LPSPI_DisableDMA(config->base, kLPSPI_TxDmaEnable | kLPSPI_RxDmaEnable);
+			/* Load dma blocks of equal length */
+			size_t dma_size = MIN(data->ctx.tx_len, data->ctx.rx_len);
+
+			if (dma_size == 0) {
+				dma_size = MAX(data->ctx.tx_len, data->ctx.rx_len);
+			}
+
+			spi_context_update_tx(&data->ctx, 1, dma_size);
+			spi_context_update_rx(&data->ctx, 1, dma_size);
```

Comment:
again, we are updating tx and rx here but only check if either of them is done? what if one is not done

Reply:
Right now, dma_size will be forced to be the same.

Reply:
that is not correct